_posts/2020-04-23-putting-the-serverless-in-graphQL-with-AppSync.md

```diff
@@ -0,0 +1,206 @@
+
+# Quick Start
+
+The nice thing about using AWS services is we can create resources with CloudFormation template. It allows you to create and provision AWS resources predictably and repeatedly.
+
+If you are new to CloudFormation, visit [Cloudformation Getting Started](https://aws.amazon.com/cloudformation/getting-started/) to get more information. Alternatively, you can create resources via AWS Management Console, it is a good starting point if you are not familar with AWS services.
+
+### Step 1: Create AWS AppSync API
+
+We will build an API for a coffee ordering application that store all the orders in the database.
+
+The following example is written using CloudFormation YAML.
+
+```yml
+Resources:
+  CoffeeShopAppSyncApi:
+    Type: AWS::AppSync::GraphQLApi
+    Properties:
+      Name: CoffeeShopAppSyncApi
+      AuthenticationType: API_Key
+
+  CoffeeShopAppSyncApiKey:
+    Type: AWS::AppSync::ApiKey
+    Properties:
+      ApiId: !GetAtt CoffeeShopAppSyncApi.ApiId
+      Description: API key for GraphQL Api
+```
+
+There are two resources in the template.
+
+Firstly, it is creating the resource for AppSyncAPI. There are few methods available for authentication type such as
+
+- API_KEY : For using API keys
+- AWS_IAM : For using AWS Identity and Access Management (IAM) permissions
+- AMAZON_COGNITO_USER_POOLS: For using an Amazon Cognito User Pool
+- OPENID_CONNECT: For using OpenID Connect Provider
+
+In this case, we are using an API key to protect the API, so we will need to create one.
+
+### Step 2: Create the Schema
+
+Now we have the API ready, we will need to create AppSync schema for the API.
+
+```yml
+CoffeeShopAppSyncSchema:
+  Type: AWS::AppSync::GraphQLSchema
+  Properties:
+    ApiId: !GetAtt CoffeeShopAppSyncApi.ApiId
+    Definition: |
+      input orderInput {
+        id: String
+        coffee: String
+        size: String
+      }
+
+      type Order {
+        id: String
+        coffee: String
+        size: String
+      }
+
+      type Mutation {
+      createOrder(input: orderInput): Order
+      }
+
+      type Query{
+      getOrder(id: String): Order
+      }
+
+      schema {
+      query: Query
+      mutation: Mutation
+      }
+```
+
+The schema we have created forms a contract between our clients and the AWS data sources ensuring the data sent back and forth adheres to the specified types.
+
+APIId is the AWS AppSync GraphQL API identifiers that you want to apply to this schema. We have created one in step 1.
+
+A GraphQL schema must have a root query and mutation type. In the schema, we have added Query with getOrder field that return an order that matches the id. Additionally, we have also added Mutation with createOrder field that create an order. This expects the input argument to match the type defined in orderInput. As a result of the successful query or mutation, the result return will match the type defined in Order.
```

Comment:
> In the schema, we have added Query

who is we? I thought it was just you (Shermayne)? SO, In the schema, I have added Query